src/index.mjs

```diff
@@ -0,0 +1,209 @@
+  schema: env.get("DATA_STORE_SCHEMA").default("polygon_parts").asString(),
+  ssl: env.get("DATA_STORE_SSL").default("DISABLE").asString(),
+};
+
+const GEOSERVER_BASE_URL = env
+  .get("GEOSERVER_BASE_URL")
+  .default("http://localhost:8080/geoserver")
+  .asString();
+
+const POLYGON_PARTS_WORKSPACE_NAME = env
+  .get("POLYGON_PARTS_WORKSPACE_NAME")
+  .default("polygon_parts")
+  .asString();
+
+const WORKSPACE_API_URL = `${GEOSERVER_BASE_URL}/rest/workspaces`;
+const DATA_STORE_API_URL = `${WORKSPACE_API_URL}/${POLYGON_PARTS_WORKSPACE_NAME}/datastores`;
+const DATA_STORE_NAME = env
+  .get("DATA_STORE_NAME")
+  .default("polygon_parts")
+  .asString();
+
+const FEATURE_TYPES_API_URL = `${DATA_STORE_API_URL}/${DATA_STORE_NAME}/featuretypes`;
+const LAYER_BODY_JSON = "./artifacts/polygonPartSpec.json";
+const MAX_FEATURES = env.get("MAX_FEATURES").default("0").asInt();
+const NUM_DECIMALS = env.get("NUM_DECIMALS").default("0").asInt();
+const LAYER_TITLE_NAME = env
+  .get("LAYER_TITLE_NAME")
+  .default("polygon_parts")
+  .asString();
+
+// will check if geoserver is up before running configuration
+while (true) {
+  try {
+    const responseFromGs = await zx.fetch(`${GEOSERVER_BASE_URL}`, {
+      method: "GET",
+      headers: {
+        Authorization: "Basic " + btoa(GEOSERVER_USER + ":" + GEOSERVER_PASS),
+      },
+    });
+    console.log(
+      `Got response from geoserver with status code: ${responseFromGs.status}`
+    );
+    assertOk(responseFromGs.status === 200);
+    break;
+  } catch (error) {
+    console.log(
+      `Failed connect to geoserver with error ${error}, will retry again`
+    );
+    await zx.sleep(5000);
+  }
+}
+
+/**
+ * 1. This stage will send api request to delete workspace if already exists
+ */
+const deleteApiParams = new URLSearchParams();
+deleteApiParams.append("recurse", true);
+
+const deleteWorkspaceResp = await zx.fetch(
+  `${WORKSPACE_API_URL}/${POLYGON_PARTS_WORKSPACE_NAME}?` + deleteApiParams,
+  {
+    method: "DELETE",
+    headers: {
+      Authorization: "Basic " + btoa(GEOSERVER_USER + ":" + GEOSERVER_PASS),
+    },
+  }
+);
+
+console.log(await deleteWorkspaceResp.text());
+assertOk(
+  deleteWorkspaceResp.status === 200 || deleteWorkspaceResp.status === 404
+);
+console.log(
+  `1. Complete validate & clean (if exists) workspace ${POLYGON_PARTS_WORKSPACE_NAME} with status code: ${deleteWorkspaceResp.status}\n`
+);
+
+await zx.sleep(1000);
+
+/**
+ * 2. This stage will send api request to create workspace
+ */
+const polygonPartWorkspaceBody = {
+  workspace: {
+    name: POLYGON_PARTS_WORKSPACE_NAME,
+  },
+};
+
+const createWorkspaceResp = await zx.fetch(WORKSPACE_API_URL, {
+  method: "POST",
+  body: JSON.stringify(polygonPartWorkspaceBody),
+  headers: {
+    Authorization: "Basic " + btoa(GEOSERVER_USER + ":" + GEOSERVER_PASS),
+    "Content-Type": "application/json",
+  },
+});
+
+console.log(await createWorkspaceResp.text());
+assertEqual(createWorkspaceResp.status, 201);
```

Comment:
what happens to this side card when assertion failed? will it be restarted?

Reply:
yes, and the readiness won't be ready